codice/ddf
# [2.26.x] DDF-6568 - Add Exiftool and configurable mappings to Tika InputTransformer

#### What does this PR do?
Adds better AVI support with Exiftool metadata keys to the MetacardCreator for the Tika based InputTransformers. The metadata mappings are also configurable by adding and specifying a custom properties file to use so that they can be overridden downstream if necessary.

#### Who is reviewing it? 

@pklinef 

#### Select relevant component teams: 

@codice/data 


#### Ask 2 committers to review/merge the PR and tag them here.
@jlcsmith 
@andrewkfiedler 

#### How should this be tested?
Install exiftool and make sure it is in the path for the  user running DDF.
Start DDF
Ingest an AVI file
Verify that the created metacard has additional data that was not normally present, such as `media.width-pixels` and `media.height-pixels`
Verify that the settings can be overridden
 - edit the `custom.system.properties` file and add the key: `tika.metadata.mapping.file` and point it to a properties file.  
 - In the properties file, add some mappings that are different than the provided mappings (e.g. description=MIME Type)
 - Update the security policy file to grant read permissions to the file for: `codeBase "file:/org.apache.tika.core/video-input-transformer/tika-input-transformer/catalog-transformer-pdf/catlog-transformer-pptx"`
 - Restart DDF
 - Ingest another AVI file and verify the new mapping is applied

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #6568 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
